components/TravelPlanForm/index.js

```diff
@@ -1,47 +1,56 @@
 import React, { useState } from 'react';
+import { useRouter } from 'next/router';
 
 const TravelPlanForm = ({ addTravelPlan }) => {
+  const router = useRouter();
+
+  // Initialize the state of the form fields
   const [plan, setPlan] = useState({
     name: '',
     startDate: '',
     endDate: '',
-    activity: ''
+    activity: '',
   });
 
+  // Handler function to update state when input fields change
   const handleChange = (event) => {
     setPlan({
       ...plan,
-      [event.target.name]: event.target.value
+      [event.target.name]: event.target.value,
     });
   };
 
-  const handleSubmit = (event) => {
+  // Handler function for form submission
+  const handleSubmit = async (event) => {
     event.preventDefault();
 
-    // Create a new plan object with the input data
-    // and an activities array with a single activity.
-    const newPlan = {
-      ...plan,
-      activities: [{
-        name: plan.activity,
-        startDate: plan.startDate,
-        endDate: plan.endDate,
-        details: '' // Fill this in with appropriate data
-      }]
-    }
-
-    // Add the new plan
-    addTravelPlan(newPlan);
-
-    // Reset the form
+    // Make a request to our API route with the plan data to save it in the database
+    const res = await fetch('/api/plans', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(plan),
+    });
+
+    const result = await res.json();
+
+    // Add the plan to the local display
+    addTravelPlan(result);
+
+    // Redirect the user to the detail page of the newly created plan
+    router.push(`/plan/${result._id}`);
+
+    // Clear the form fields
     setPlan({
```

Comment:
```suggestion
    setPlan(initialPlanState);
```
And remove the lines below.